pkg/tf2pulumi/convert/eject_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/stretchr/testify/assert"
+)
+
+// Asserts that err is nil and if not stops the test
+func stopOnError(t *testing.T, err error) {
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+}
+
+type testLoader struct {
+	path string
+}
+
+func (l *testLoader) LoadPackage(pkg string, version *semver.Version) (*schema.Package, error) {
+	schemaPath := pkg
+	if version != nil {
+		schemaPath += "-" + version.String()
+	}
+	schemaPath = filepath.Join(l.path, schemaPath) + ".json"
+
+	schemaBytes, err := os.ReadFile(schemaPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var spec schema.PackageSpec
+	err = json.Unmarshal(schemaBytes, &spec)
+	if err != nil {
+		return nil, err
+	}
+
+	schemaPackage, diags, err := schema.BindSpec(spec, l)
+	if err != nil {
+		return nil, err
+	}
+	if diags.HasErrors() {
+		return nil, diags
+	}
+
+	return schemaPackage, nil
+}
+
+func (l *testLoader) LoadPackageReference(pkg string, version *semver.Version) (schema.PackageReference, error) {
+	schemaPackage, err := l.LoadPackage(pkg, version)
+	if err != nil {
+		return nil, err
+	}
+	return schemaPackage.Reference(), nil
+}
+
+func TestEject(t *testing.T) {
+	// Test framework for eject
+	// Each folder in testdata has a pcl folder, we check that if we convert the hcl we get the expected pcl
+	testDir, err := filepath.Abs(filepath.Join("testdata"))
+	stopOnError(t, err)
+	infos, err := os.ReadDir(testDir)
+	stopOnError(t, err)
+
+	tests := make([]struct {
+		name string
+		path string
+	}, 0)
+	for _, info := range infos {
+		// Skip the "schemas" directory, that's for test schemas not for tests themselves
+		if info.IsDir() && info.Name() != "schemas" {
+			tests = append(tests, struct {
+				name string
+				path string
+			}{
+				name: info.Name(),
+				path: filepath.Join(testDir, info.Name()),
+			})
+		}
+	}
+
+	loader := &testLoader{path: filepath.Join(testDir, "schema")}
```

Comment:
Since we skipped a directory called `schemas` that include the test schemas, should this also be `schemas`?

Reply:
hah yes! Adding a test that actually needed a schema was going to be the next thing I added and I would of stumbled on this then.